pos_payment_terminal/static/src/js/models.js

```diff
@@ -0,0 +1,125 @@
+/*
+    POS Payment Terminal module for Odoo
+    Copyright (C) 2014-2016 Aurélien DUMAINE
+    Copyright (C) 2014-2016 Akretion (www.akretion.com)
+    Copyright (C) 2019 ACSONE SA/NV
+    @author: Aurélien DUMAINE
+    @author: Alexis de Lattre <alexis.delattre@akretion.com>
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+*/
+
+odoo.define('pos_payment_terminal.models', function (require) {
+    "use strict";
+
+    var models = require('point_of_sale.models');
+
+    var _orderproto = models.Order.prototype;
+    models.Order = models.Order.extend({
+        get_paymentline: function(id){
+            var paymentlines = this.paymentlines.models;
+            for(var i = 0; i < paymentlines.length; i++){
+                if(paymentlines[i].cid === id){
+                    return paymentlines[i];
+                }
+            }
+            return null;
+        },
+        get_payment_line_by_transaction: function(transaction_id){
+            // This will retrieve the payment line by transaction reference
+            self = this;
+            var payment_lines = self.get_paymentlines();
+            for (var id in payment_lines){
+                var payment_line = payment_lines[id]
+                if (payment_line.terminal_transaction_id === undefined){
+                    continue;
+                }
+                if (transaction_id === payment_line.terminal_transaction_id){
+                    return payment_line;
+                }
+            }
+            return;
+        },
+        show_validate_button: function(){
+            var show = true;
+
+            if (self.pos.config.protect_automatic_payment){
+                var payment_lines = self.get_paymentlines();
+                for (var id in payment_lines){
+                    var payment_line = payment_lines[id]
+                    if payment_line.in_transaction{
```

Comment:
I don't think we need `in_transaction`: it should be equivalent to (transaction_id is not null and transaction_success is null).